qutip/solver/ode/qutip_integrator.py

```diff
@@ -0,0 +1,119 @@
+from ..integrator import (IntegratorException, Integrator, sesolve_integrators,
+                          mesolve_integrators, mcsolve_integrators,
+                          add_integrator)
+from .explicit_rk import Explicit_RungeKutta
+from ..options import SolverOdeOptions
+import numpy as np
+from qutip import data as _data
+
+
+__all__ = ['IntegratorVern', 'IntegratorDiag']
+
+
+class IntegratorVern(Integrator):
+    """
+    Qutip's implementation of Verner 'most efficient' Runge-Kutta method
+    of order 7 and 9. It's a Runge-Kutta method with variable steps and dense
+    output.
+    Use qutip's Data object for the state, allowing sparse or gpu states.
+    [http://people.math.sfu.ca/~jverner/]
+    """
+    integrator_options = {
+        'atol': 1e-8,
+        'rtol': 1e-6,
+        'nsteps': 1000,
+        'first_step': 0,
+        'max_step': 0,
+        'min_step': 0,
+        'interpolate': True,
+        'method': 'vern7'
+    }
+    support_time_dependant = True
+    supports_blackbox = True
+
+    def _prepare(self):
+        self._ode_solver = Explicit_RungeKutta(self.system, **self.options)
+        self.name = "qutip " + self.options['method']
+
+    def get_state(self, copy=True):
+        state = self._ode_solver.y
+        return self._ode_solver.t, state.copy() if copy else state
+
+    def set_state(self, t, state):
+        self._ode_solver.set_initial_value(state, t)
+
+    def integrate(self, t, copy=True):
+        self._ode_solver.integrate(t, step=False)
+        self._check_failed_integration()
+        return self.get_state(copy)
+
+    def mcstep(self, t, copy=True):
+        self._ode_solver.integrate(t, step=True)
+        self._check_failed_integration()
+        return self.get_state(copy)
+
+    def _check_failed_integration(self):
+        if self._ode_solver.successful():
+            return
+        messages = {
```

Comment:
Perhaps the messages can made accessible directly on the ODE solver? I know scipy doesn't to this, but perhaps we can make ours nicer?